### PR TITLE
uri_string:normalize/2 may return error()

### PR DESCRIPTION
**Pull request by essen**
~~Also remove an unnecessary try/catch in uri_string:normalize/1.~~

### CODE REVIEW

**Comment by peterdmv** [comment]
Can you please rebase also this PR to maint?

**Comment by essen** [comment]
I did it on top of maint I just messed up when opening the PR.

File: lib/stdlib/src/uri_string.erl [hunk 1]
```diff
@@ -297,16 +297,14 @@
       NormalizedURI :: uri_string()
                      | error().
 normalize(URIMap) ->
-    try normalize(URIMap, [])
-    catch
-        throw:{error, Atom, RestData} -> {error, Atom, RestData}
-    end.
+    normalize(URIMap, []).
```
Comment (peterdmv) [inline lib/stdlib/src/uri_string.erl]: uri_string_SUITE:normalize_negative was failing in our CI due to this change.
```
{error,invalid_utf8,<<0,0,0,246>>} = uri_string:normalize("//%00%00%00%F6")
```
To make it more complicated I found problems even with the original code. The following examples trigger uncaught exceptions:
```
uri_string:normalize("//%00%00%00%F6", [return_map])
uri_string:normalize(#{host => "%00%00%00%F6",path => []}, [return_map]).
```
So if we keep the scope of this PR, I would propose to keep the try catch.

Comment (essen) [reply lib/stdlib/src/uri_string.erl]: Seems that `uri_string:normalize("/%XX")` would also trigger a throw.

I'll put the try/catch back.

**Comment by peterdmv** [comment]
Thanks for the contribution!

**Pull request merged**